stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -457,6 +457,27 @@ end
 
 *(A::HermOrSym, B::HermOrSym) = A * copyto!(similar(parent(B)), B)
 
+function dot(x::AbstractVector, A::RealHermSymComplexHerm, y::AbstractVector)
+    require_one_based_indexing(x, y)
+    (length(x) == length(y) == size(A, 1)) || throw(DimensionMismatch())
+    Δ = A.uplo == 'U' ? UpperTriangular(A) : LowerTriangular(A)
```

Comment:
This makes a copy of the data (at least in the complex case, to zero the imaginary parts on the diagonal).   It is also type-unstable.

Seems like you should really just write out the loops:
```
data = A.data
if A.uplo == 'U'
    r = zero(eltype(x)) * zero(eltype(A)) * zero(eltype(y))
    @inbounds for j = 1:length(y)
        r += dot(x[j], real(data[j,j]), y[j])
        @simd for i = 1:j-1
            Aij = data[i,j]
            r += dot(x[i], Aij, y[j]) + dot(x[j], conj(Aij), y[i])
        end
    end
else # A.uplo == 'L'
    ...
end
```
Not sure a specialized `x === y` case is worth it?